docs/standards.md

```diff
@@ -197,15 +198,140 @@ Things to take note of:
 
 ### Test files
 
-`TODO`
+We use [Mocha](https://mochajs.org/) with [Chai](https://www.chaijs.com/) to test our routes and services. These test files are located in the `tests` folder, and are named `<X>.test.js` or `<X>.spec.js`. It is important to test both succcess and fail cases. For example, testing account retrieval may include scenarios of:
+  * Failure due to authentication
+  * Failure due to authorization
+  * Failure due to the account not existing
+  * Success case for a user
+  * Success case for an admin
+The code for this example can be found in [account.test.js](../tests/account.test.js).
+
+We repopulate the test server before each test to ensure consistency. [setup.spec.js](../tests/setup.spec.js) contains the code for that. The `storeAll` and `dropAll` functions call test util functions that store and drop specific collections. For example, `account.test.util.js` contains the code a `storeAll` function that inserts all the test account documents into the test database.
+
+#### Util files and Test Database Population
+
+##### Motivation
+
+We wanted to have a scalable way to create new entities that properly reference each other. The biggest challenge lies in creating enough accounts that can be properly referenced during specific tests. 
+
+##### Util.js
+
+Account.util.js contains the test data for accounts. When an account is created, it is placeed within an object whose name references the type of account it will be (ex: Hacker, Sponsor, etc). The object has keys that further define the state of the account. The keys are: `new` for new accounts, `stored` for accounts that already exist, and `invalid` for invalid accounts. Additionally, there are 2 more keys inside `stored` for hacker accounts, which are `team` and `noTeam` for accounts linked to hackers that are on a team, or not on a team respectively.
+
+```javascript
+let hackerAccounts = {
+    new: createNAccounts(10, {
+        "accountType": Constants.HACKER,
+        "confirmed": true,
+    }),
+    stored: {
+        team: createNAccounts(10, {
+            "accountType": Constants.HACKER,
+            "confirmed": true,
+        }),
+        noTeam: createNAccounts(10, {
+            "accountType": Constants.HACKER,
+            "confirmed": true,
+        }),
+    },
+    invalid: createNAccounts(10, {
+        "accountType": Constants.HACKER
+    })
+};
+```
+
+In this example the `new` accounts are accounts that exist, but the hacker objects have not been created. The `stored.team` accounts are those linked to a hacker object that is in a team. The `stored.noTeam` accounts link to  a hacker that is not in a team. The `invalid` accounts are created accounts that are linked to a hacker object that does not fit with the Hacker schema. The invalid accounts are used to test fail cases. The value for each key is an array of account objects. 
```

Comment:
Add on the following sentence to this one:

> The `invalid` accounts are created accounts that are linked to a hacker object that does not fit with the Hacker schema. 

"These accounts are only invalid because the hacker object is malformed."